llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1744,11 +1749,14 @@ ModulePassManager PassBuilder::buildThinLTODefaultPipeline(
     MPM.addPass(GlobalDCEPass());
     return MPM;
   }
-
-  // Add the core simplification pipeline.
-  MPM.addPass(buildModuleSimplificationPipeline(
-      Level, ThinOrFullLTOPhase::ThinLTOPostLink));
-
+  if (!UseCtxProfile.empty()) {
+    MPM.addPass(
```

Comment:
this change reduces the number of simplification passes (pre and post inliner) run. What is the reason?

Reply:
OR the intention is to do a quick e2e integration?

Reply:
Right now yes, it's just to integrate. The next step plan is to run some of the module simplification after all IPO and flattened profile lowering happens, but I want to see first what the more meaningful to do that is - special case through `buildModuleSimplificationPipeline` or have them spelled out separately. I'll evaluate this experimentally.